source/beamline/AMMotorGroup.h

```diff
@@ -27,153 +27,831 @@ along with Acquaman.  If not, see <http://www.gnu.org/licenses/>.
+	  */
+	QString rotationName() const;
+
+	/*!
+	  * The current translational position of this axis. If the axis cannot
+	  * translate, a value of 0 is returned.
+	  */
+	double currentTranslationPosition() const;
+
+	/*!
+	  * The current rotational position of this axis. If the axis cannot rotate,
+	  * a value of 0 is returned.
+	  */
+	double currentRotationPosition() const;
+
+	/*!
+	  * The units in which the translations of the axis are measured.
+	  */
+	QString translationPositionUnits() const;
+
+	/*!
+	  * The units in which the rotations of the axis are measured.
+	  */
+	QString rotationPositionUnits() const;
+
+	/*!
+	  * The smallest position the rotational motor can take.
+	  */
+	double minRotationPosition() const;
+
+	/*!
+	  * The smallest position the translational motor can take.
+	  */
+	double minTranslationPosition() const;
+
+	/*!
+	  * The lasgest position the rotational motor can take.
+	  */
+	double maxRotationPosition() const;
+
+	/*!
+	  * The lasrgest position the translation motor can take.
+	  */
+	double maxTranslationPosition() const;
+
+	/*!
+	  * Whether this axis is currently moving (translating or rotating).
+	  */
+	bool isMoving() const;
+
+	/*!
+	  * Whether this axis is currently translating.
+	  */
+	bool isTranslating() const;
+
+	/*!
+	  * Whether this axis is currently rotating.
+	  */
+	bool isRotating() const;
+
+	/*!
+	  * Whether this axis can make a translational movement. Checks to ensure that
+	  * a motion is associated with translational movements for this axis, that
+	  * a motor is associated with that motion, and that the motor is connected.
+	  */
+	bool canTranslate() const;
+
+	/*!
+	  * Whether this axis can make a rotational movement. Checks to ensure that
+	  * a motion is associated with rotational movements for this axis, that
+	  * a motor is associated with that motion, and that the motor is connected.
+	  */
+	bool canRotate() const;
+
+	/*!
+	  * Instructs the translation motor to move to the provided position.
+	  * \param position ~ The position to which the translation motor will attempt
+	  * to move.
+	  */
+	void setTranslatePosition(double position);
+
+	/*!
+	  * Instructs the rotation motor to move to the provided position.
+	  * \param position ~ The position to which the rotation motor will attempt
+	  * to move.
+	  */
+	void setRotatePosition(double position);
+
+	/*!
+	  * Sets the override values for the position units for the translational motions.
+	  * \param positionUnits ~ The position units to associate with the translational
+	  * movements.
+	  */
+	void setTranslationPositionUnits(const QString& positionUnits);
+
+	/*!
+	  * Sets the override values for the position units for the rotational motions.
+	  * \param positionUnits ~ The position units to associate with the rotational
+	  * movements.
+	  */
+	void setRotationPositionUnits(const QString& positionUnits);
+
+	/*!
+	  * Stops motions of the translation motor.
+	  */
+	void stopTranslation();
```

Comment:
In general, all of the action items for a motor group object should be slots.  That way (in particular with stop) there is a very straight forward "button click -> mapped directly to stop".  It also makes it more convenient for programmatic calls more succinct rather than having to make their own slot just call stop or move.  For all versions of move/stop at all levels, they should likely be modified into slots.